### ISSUE TITLE
Support cmd tool --version

### ISSUE THREAD

**Issue by zrss** [google/go-containerregistry#272]
## Expected

ko (and the rest tools) --version can print the version info of ko

**Comment by zrss** [comment]
like the `go dep` ?

[https://github.com/golang/dep#installation](https://github.com/golang/dep#installation)

install from the release

```
root@zrss-dev-01:~/xvde/goproject/src/github.com/knative/build(master)# dep version
dep:
 version     : v0.5.0
 build date  : 2018-07-26
 git hash    : 224a564
 go version  : go1.10.3
 go compiler : gc
 platform    : linux/amd64
 features    : ImportDuringSolve=false
```

and install from the go get

```
root@zrss-dev-01:~# dep version
dep:
 version     : devel
 build date  : 
 git hash    : 
 go version  : go1.10.4
 go compiler : gc
 platform    : linux/amd64
 features    : ImportDuringSolve=false
```

**Comment by ImJasonH** [comment]
`docker` does a similar thing. I was wondering if you could give us some pointers to how it's implemented. Looks like `dep`'s is [here](https://github.com/golang/dep/blob/master/cmd/dep/version.go).

The `commitHash` seems to be inserted from a release shell script [here](https://github.com/golang/dep/blob/3807092e689e29a36a6be09a6d66b09488b6ac16/hack/build-all.bash#L27).

**Comment by ImJasonH** [comment]
`docker` seems to do a similar thing, [here](https://github.com/moby/moby/blob/3d9adede1320495be9dd4fb4ce1eee2c180b7fe1/hack/make/.go-autogen).

**Comment by zrss** [comment]
but currently, we don't have a official `ko` release channel, right ? so it seems it can't be done by `go build ...`, maybe i can find something in `golang` upstream

**Comment by zrss** [comment]
`dep` has two choice to get it, one way get the pre-build release, the other way get it from `go get`, u can see the `go get` lack of the commit and build date messages ...

**Comment by ImJasonH** [comment]
That's correct, we haven't cut a real release of `ko` yet (or `crane`). It would be nice to do this before we do a release of either.

**Comment by zrss** [comment]
fair enough, that's the reason why i wonder ko lack of the version info ... but it seems it will be ready when we cut a release and add some release scripts

**Comment by bobcatfish** [comment]
Bumping this, in https://github.com/tektoncd/pipeline we would like to be able to depend on a specific version of `ko`, in https://github.com/tektoncd/pipeline/pull/605 we're going to be telling our contributors to install a version of `ko` older than a certain commit, it would be great to be able to pin to a version.
